template/downstream.c.erb

```diff
@@ -53,15 +61,11 @@ void btx_downstream_push_message(
       message_iterator_private_data->common_data->usr_data, message);
   else
     _btx_push_message(message_iterator_private_data, message);
+  <% else %>
+  _btx_push_message(message_iterator_private_data, message);
+  <% end %>
```

Comment:
What do you think about this refactoring?

```ruby
  <% if options[:enable_on_downstream_callback] %>
  on_downstream_message_callback_f *p =
      (on_downstream_message_callback_f *)message_iterator_private_data
          ->common_data->on_downstream_message_callback;
  if (p) {
    p(((void *)message_iterator_private_data),
      message_iterator_private_data->common_data->usr_data, message);
    return;
  }
  <% end %>
  _btx_push_message(message_iterator_private_data, message);
```

Reply:
Yeah shorter! Lets go with that, I think I didn't liked it at first because it mike 2 kind of conditional (static and dynamic) but it's shorted indeed!